### DISCUSSION TITLE
How can I make a squash?

### DISCUSSION THREAD

**Discussion by elC0mpa** [NeogitOrg/neogit#1177] [General]
Firs of all, thanks for this great NeoVIM plugin. I have been using Github Desktop for a while but decided to switch to NeoGit and everything is ok so far. I have a question
![image](https://github.com/NeogitOrg/neogit/assets/53531665/f9cf60b5-70a4-4202-bc3b-62ff7b8f81c4)
I have the two commits shown in the above picture  and I want to squash them before pushing to origin. As you can see in the image I can select both commits and squash them to later make the push of a single commit. 
However when I try to squash inside neogit it gives me the following output: 'No changes to commit'
Any idea on how to achieve this? One question, which is the difference between squash and instant squash?

Thanks in advance

**Comment by elC0mpa** [comment]
Another question, when pressed P to open the Push popup, how can I specify it to be a force push? I know it is done by typing -f, but how could I achieve it from NeoGit?

**Reply by CKolkey** [reply]
Easy - you type `-f` to do "--force-with-lease" and "-F" to do "--force" :) All the characters on the left indicate how to toggle that switch/option. Some have a `=` prefix, some `-`.
 
<img width="309" alt="Screenshot 2024-02-15 at 21 39 20" src="https://github.com/NeogitOrg/neogit/assets/7228095/66012f91-7124-4439-b02e-2806bae52b4b">

**Reply by elC0mpa** [reply]
Thanks, what about the squash question? The original one?

**Reply by CKolkey** [reply]
Answered below :)

**Comment by CKolkey** [comment]
> I have the two commits shown in the above picture and I want to squash them before pushing to origin. As you can see in the image I can select both commits and squash them to later make the push of a single commit. However when I try to squash inside neogit it gives me the following output: 'No changes to commit' Any idea on how to achieve this? 

You would open an interactive rebase `ri`, then use `s` on the commits you want to squash, and finally save/close the buffer via `<c-c><c-c>`.

> One question, which is the difference between squash and instant squash?

Squash will take your staged changes and create a `squash! ...` commit, and instant squash will do that and then rebase that commit too. Give it a try! You can always reset your current branch via `bX` (branch popup, reset)
